### PR TITLE
Implement database's record population

### PR DESCRIPTION
**Pull request by oskarcieslik**

### CODE REVIEW

File: src/kakapo.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
...
-export * from './router';
-export * from './server';
-export * from './response';
\ No newline at end of file
+export { Database } from './database';
+export { Router } from './router';
+export { Server } from './server';
+export { Response } from './response';
+
+export class Kakapo {
```
Comment (zzarcon) [inline src/kakapo.js]: This class have to be removed

File: tests/specs/database_spec.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  });
+
+  test('DB # uuid', assert => {
+    const db = new Database();
+
+    db.register('user', userFactory);
+
+    assert.equal(db.uuid('user'), 0, 'Returns 0 as first identifier.');
+    assert.equal(db.uuid('user'), 1, 'Returns bigger identifier than before.');
+
```
Comment (zzarcon) [inline tests/specs/database_spec.js]: Can you add here a test checking the `uuid` with a different entity? That way we'll ensure that the id is unique per entity ^^

Comment (oskarcieslik) [inline tests/specs/database_spec.js]: On it! :)

File: src/database.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  factoryFor(collectionName) {
+    return this.factories[collectionName];
+  }
+
+  filter(collectionName, conditions) {
+    this.checkFactoryPresence(collectionName);
+    return _.filter(this.store[collectionName], conditions);
+  }
+
+  find(collectionName, conditions) {
```
Comment (zzarcon) [inline src/database.js]: Should we `checkFactoryPresence` here?

**Pull request merged**